src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2576,6 +2596,45 @@ static void start_retriable_subchannel_batches(void* arg, grpc_error* ignored) {
 // LB pick
 //
 
+// The callback to intercept trailing metadata if retries is not enabled
+static void recv_trailing_metadata_ready_for_lb(void* arg, grpc_error* error) {
+  grpc_call_element* elem = static_cast<grpc_call_element*>(arg);
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  if (calld->pick.recv_trailing_metadata != nullptr) {
+    *calld->pick.recv_trailing_metadata =
+        calld->recv_trailing_metadata_op_batch->payload
+            ->recv_trailing_metadata.recv_trailing_metadata;
+  }
+  GRPC_CLOSURE_SCHED(
```

Comment:
Just had another thought here: An LB policy implementation may have its own reasons for needing to process the trailing metadata asynchronously, in which case the second approach I mentioned above (requiring the call into the LB policy to be synchronous) wouldn't really work.  So I think we probably do want to just treat it like any filter that intercepts a callback and make it be responsible for chaining to the previous callback when it's done.  This is unfortunate, because it does broaden the LB policy API in a way that seems counter to our goal of making it a public API, but I don't see any reasonable way around it, at least for now. :(